python-threatexchange/threatexchange/exchanges/impl/fb_threatexchange_api.py

```diff
@@ -323,19 +339,132 @@ def report_false_positive(
+                for tx_s_type in s_types:
+                    s_type_specific_signal_str = (
+                        tx_s_type.normalize_fb_threatexchange_indicator(
+                            type_str, signal_str, tag
+                        )
+                    )
+                    s_type = t.cast(t.Type[SignalType], tx_s_type)
+                    inner = ret.get(s_type)
+                    if inner is None:
+                        inner = {}
+                        ret[s_type] = inner
+                    to_insert = _merge_record_for_signal_type(
+                        metadata, tag, inner.get(s_type_specific_signal_str)
+                    )
+                    if to_insert is not None:
+                        inner[s_type_specific_signal_str] = to_insert
+
+        return ret
+
+
+def _merge_record_for_signal_type(
+    tx_record: FBThreatExchangeIndicatorRecord,
+    tag: t.Optional[str],
+    existing: t.Optional[FBThreatExchangeIndicatorRecord],
+) -> t.Optional[FBThreatExchangeIndicatorRecord]:
+    if tag is not None:
+        applicable_opinions = [
+            o for o in tx_record.opinions if any(t in tag for t in o.tags)
+        ]
+        if not applicable_opinions:
+            return None
+        if len(applicable_opinions) != len(tx_record.opinions):
+            tx_record = FBThreatExchangeIndicatorRecord(applicable_opinions)
+    if existing is not None:
+        existing.merge(tx_record)
+        return None
+    return tx_record
+
+
+def _indicator_applies(
+    u: ThreatUpdateJSON,
+    type_mapping: t.Mapping[
+        str,
+        t.Mapping[
+            t.Optional[str], t.Sequence[t.Type[HasFbThreatExchangeIndicatorType]]
+        ],
+    ],
+) -> t.Optional[FBThreatExchangeIndicatorRecord]:
+    """Based on the available signal types, return a record"""
+    potential_signal_type = type_mapping.get(u.threat_type)
+    if potential_signal_type is None:
+        return None
+    indicator = FBThreatExchangeIndicatorRecord.from_threatexchange_json(u)
+    if indicator is None:
+        return None
+    if None in potential_signal_type:
+        return indicator
+    if any(
+        tag in potential_signal_type
+        for opinion in indicator.opinions
+        for tag in opinion.tags
+    ):
+        return indicator
+    return None
+
 
 def _make_indicator_type_mapping(
     supported_signal_types: t.Sequence[t.Type[SignalType]],
-) -> t.Dict[str, t.Type[SignalType]]:
-    # TODO - We can clobber types that map into multiple
-    type_mapping: t.Dict[str, t.Type[SignalType]] = {}
+) -> t.Mapping[
+    str,
+    t.Mapping[t.Optional[str], t.Sequence[t.Type[HasFbThreatExchangeIndicatorType]]],
+]:
+    """
+    Based on the given signal types, create a map for converting ThreatIndicators
+    """
+    ret: t.DefaultDict[
+        str,
+        t.DefaultDict[
+            t.Optional[str], t.List[t.Type[HasFbThreatExchangeIndicatorType]]
+        ],
+    ] = defaultdict(lambda: defaultdict(list))
```

Comment:
non-blocking: calling [] on mapping will still create empty records. Just having t.Mapping does not guarantee immutability.

Reply:
Add the map example to doc string.